.changeset/heavy-kings-add.md

```diff
@@ -0,0 +1,22 @@
+---
+'@remote-ui/rpc': patch
+---
+
+Throw better errors on the RESULT and FUNCTION_RESULT methods
+
+```js
+case RESULT: {
+    const [callId, method] = data[1];
+
+    try {
+        callIdsToResolver.get(callId)!(...data[1]);
+        callIdsToResolver.delete(callId);
+    } catch (error) {
+        const {message} = error as Error;
+        throw new Error(
+        `Error in result listener. Method: ${method} Error: ${message}`,
+        );
+    }
+    break;
+}
+```
```

Comment:
This ends up constructing a changelog, and I don’t think consumers typically expect to see comments on implementation details like these in a changelog. If we agree that a custom error is the path forward, I think it's clearer to say something like the:

```md
An endpoint will trigger an uncaught promise rejection with a `<INSERT_ERROR_NAME>Error` error when it receives messages to call a function that is no longer registered.
```